### PR TITLE
Override --connection if specified through CLI arguments

### PR DESCRIPTION
**Pull request by darnfish**
This PR aims to fix issues where the Knexfile would be chosen as the preferred argument provider even if a `--connection` arg was provided

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -48,11 +48,17 @@ async function initKnex(env, opts) {
...
-    opts,
-    env.configuration,
-    env.configPath
-  );
+  const resolvedConfig = resolveEnvironmentConfig(opts, env.configuration);
+
+  // override connection, client and migrationsTableName
+  // if specified through cmd line arguments
+
+  if (opts.connection) resolvedConfig.connection = opts.connection;
```
Comment (kibertoad) [inline bin/cli.js:56]: How do you feel about resolvedConfig.x = opts.x || resolvedConfig style?

Comment (BossCalifornia) [reply bin/cli.js:56]: I can fix that !!

Comment (BossCalifornia) [reply bin/cli.js]: Done in 0780b70

**Comment by kibertoad** [comment]
@BossCalifornia Lots of existing tests are failing now.

**Comment by BossCalifornia** [comment]
@kibertoad I'll look into it soon 👍

**Comment by kibertoad** [comment]
@darnfish Do you need any help with this?

**Comment by OlivierCavadenti** [comment]
New PR here : https://github.com/knex/knex/pull/5006

**Pull request closed without merging**